daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1458,9 +1463,11 @@ object Main {
     data: InputSourceDataInputStream,
     saxContentHandler: ContentHandler): ParseResult = {
     val saxXmlRdr = processor.newXMLReaderInstance
-    saxXmlRdr.setContentHandler(saxContentHandler)
+    // SAX_NAMESPACE_PREFIXES_FEATURE is needed to preserve nil attributes with EXI
+    saxXmlRdr.setFeature(XMLUtils.SAX_NAMESPACE_PREFIXES_FEATURE, true)
     saxXmlRdr.setProperty(XMLUtils.DAFFODIL_SAX_URN_BLOBDIRECTORY, blobDir)
     saxXmlRdr.setProperty(XMLUtils.DAFFODIL_SAX_URN_BLOBSUFFIX, blobSuffix)
+    saxXmlRdr.setContentHandler(saxContentHandler)
```

Comment:
Are no changes needed to the `unparseWIthSax` function for schema unaware exi unparsing? Doesn't the `EXISource` `XMLReader` need to know the grammar to be able to deserialize the EXI file?

 Or does it just serialize the entire grammar into the EXI file? Seems like that would defeat the purpose of schema awareness so I imagine that's not the case?

Reply:
This will be addressed in my next commit.  Going to modify things to only create one EXIFactory instance that can be used for both parsing and unparsing.